### PR TITLE
parallel in crystal alignment

### PR DESCRIPTION
**Pull request by abuts**
The PR fixes Re #1721 by disabling parallel multifit in crytal_alignment. It also adds the test, which demonstrates that parallel multifit fails for crytal_alignment case.

### CODE REVIEW

File: _test/test_multifit_herbert/test_multifit_1.m [hunk 1]
```diff
@@ -324,5 +331,45 @@
...
+            kk = kk.set_fun (@reciprocal_space_deviation, {pars,rlu_expected});
+            kk = kk.set_options ('list', 0);
+            kk = kk.set_options ('fit',[1e-4,50,-1e-6]);
+
+            [distance,fitpar] = kk.fit();
+            assertEqualWithSave(obj,distance,'tol',1.e-4);
+            assertEqualWithSave(obj,fitpar,'tol',1.e-3);
+            clear clOb;
+
+            skipTest('Re #1724 This fit fails and should be fixed as part of the ticket #1724')
```
Comment (cmarooney-stfc) [inline _test/test_multifit_herbert/test_multifit_1.m:366]: this suggests that the fix is going to be enabling parallel multifit. But the fix in this PR is to disable parallel multifit. At the least this message needs modifying

File: horace_core/lattice_functions/refine_crystal.m [hunk 2]
```diff
@@ -167,9 +167,12 @@
 end
 
 kk = multifit (vcryst0_3, zeros(3*nv,1), 0.01*ones(3*nv,1));
+% function to caclulate distance between actual and expected bragg peak
+% positions in reciprocal space.
 kk = kk.set_fun (@reciprocal_space_deviation, {pars,rlu_expected}, pfree, pbind);
 kk = kk.set_options ('list', 0);
 kk = kk.set_options ('fit',[1e-4,50,-1e-6]);
+clOb = set_temporary_config_options('hpc_config','parallel_multifit',false);
```
Comment (cmarooney-stfc) [inline horace_core/lattice_functions/refine_crystal.m:175]: this appears to be the only part of this PR that actually makes a change; the rest is refactoring and testing. There is no comment here to say - parallel multifit disabled here due to ticket 1724. The reader should be aware why this line is here and be able to trace it through the system. Please add the comment

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]

File: horace_core/lattice_functions/refine_crystal.m (hunk 2 above)
Comment (oerc0122) [reply horace_core/lattice_functions/refine_crystal.m:175]: `set_temporary_config_options` is only available during testing (it's in `test/common_functions`. It cannot be used outside.

File: _test/test_multifit_herbert/test_multifit_1.m (hunk 1 above)
Comment (abuts) [reply _test/test_multifit_herbert/test_multifit_1.m:366]: parallel multifit is disabled in `crystal_alignment`. This test shows that parallel multifit fails for a task, similar to `crystal_aligmnent` but not actually it. You may see that no `crystal_alignment` is invoked here.

File: horace_core/lattice_functions/refine_crystal.m (hunk 2 above)
Comment (oerc0122) [reply horace_core/lattice_functions/refine_crystal.m:175]: Or you have moved it in this PR without saying that anywhere

File: _test/test_change_crystal/test_change_crystal_bragg_coarse.m [hunk 3]
```diff
@@ -632,6 +634,36 @@ function test_rotmat_recovery(~)
 
             assertElementsAlmostEqual(corr.rotmat,rotmat_rec);
         end
+        function test_corrections_work_in_parallel(obj)
+            % testing if multifit works in parallel (actually no parallel 
+            % should be invoked) for this kind of task
```
Comment (oerc0122) [inline _test/test_change_crystal/test_change_crystal_bragg_coarse.m:639]: I do not understand this comment which contradicts itself.

File: horace_core/lattice_functions/refine_crystal.m (hunk 2 above)
Comment (abuts) [reply horace_core/lattice_functions/refine_crystal.m:177]: I've moved it into `herbert/utilities`

File: _test/test_change_crystal/test_change_crystal_bragg_coarse.m (hunk 3 above)
Comment (abuts) [reply _test/test_change_crystal/test_change_crystal_bragg_coarse.m:639]: multifit should work regardless of parallel_multifit option is enabled or not. How it is achieved -- up to ticket Re #1724

Comment (oerc0122) [reply _test/test_change_crystal/test_change_crystal_bragg_coarse.m:639]: The comment still doesn't make any sense to me.

Comment (abuts) [reply _test/test_change_crystal/test_change_crystal_bragg_coarse.m]: modified

**Pull request merged**